automotive/src/main/java/au/com/shiftyjelly/pocketcasts/AutomotiveSettingsFragment.kt

```diff
@@ -49,15 +77,55 @@ class AutomotiveSettingsFragment : Fragment() {
 
     private fun onProfileAccountButtonClicked(loggedIn: Boolean) {
         if (loggedIn) {
-            val fragment = AccountDetailsFragment.newInstance()
-            (activity as? AutomotiveSettingsActivity)?.addFragment(fragment)
+            openSettingsFragment()
         } else {
-            signIn()
+            launch {
+                signIn()
+            }
         }
     }
 
-    fun signIn() {
-        val loginIntent = Intent(activity, AccountActivity::class.java)
-        startActivity(loginIntent)
+    private suspend fun signIn() {
+        // ask the user if they want to clear their data before they sign in
+        if (isAccountUsed()) {
+            val context = context ?: return
+            val themedContext = if (Util.isAutomotive(context)) ContextThemeWrapper(context, CR.style.Theme_Car_NoActionBar) else context
+            val builder = AlertDialog.Builder(themedContext)
+            builder.setTitle(getString(LR.string.profile_clear_data_question))
+                .setMessage(getString(LR.string.profile_clear_data_would_you_also_like_question))
+                .setPositiveButton(getString(LR.string.sign_in)) { _, _ -> openSignInActivity() }
+                .setNegativeButton(getString(LR.string.profile_clear_data)) { _, _ ->
+                    userManager.signOutAndClearData(
+                        playbackManager = playbackManager,
+                        upNextQueue = upNextQueue,
+                        playlistManager = playlistManager,
+                        folderManager = folderManager,
+                        searchHistoryManager = searchHistoryManager,
+                        episodeManager = episodeManager,
+                        wasInitiatedByUser = true
```

Comment:
I wonder if `wasInitiatedByUser` should be `false` here. The user is initiating the action here, but they're not trying to sign out. One of the things this `wasInitiatedByUser` variable is used for is [to identify in our analytics whether the user was attempting to sign out](https://github.com/Automattic/pocket-casts-android/blob/5c6a321c9115b6290e979bbccc50fb0c30a1398f/modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/user/UserManager.kt#L123). 

Admittedly this is a bit of a moot point at this time since we're not sending any analytics to Tracks from Android Automotive yet.

Also, if you change `wasInitiatedByUser` to be `false`, you may need to do something like add a call to `settings.setFullySignedOut(false)` to get past [this if-check](https://github.com/Automattic/pocket-casts-android/blob/5c6a321c9115b6290e979bbccc50fb0c30a1398f/modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/user/UserManager.kt#L108) in the `signOut` method.

The `settings.getFullySIgnedOut` value is a relatively recent addition I did largely to avoid extra signout calls to analytics (we were making a call to signout every time a user who was not signed in started the app). Maybe you have some ideas for how to better accomplish that because it does seem to be creating some undesirable complexity here. 🤔

Reply:
I wonder if it would be easier to understand if we named the parameter something like `userAttemptingToSignOut`. 🤔 

I was trying to reuse the same logic but I think I have made it more complicated. Thanks for picking this up, I have committed a fix to set `wasInitiatedByUser` to `false`.

Reply:
> I wonder if it would be easier to understand if we named the parameter something like `userAttemptingToSignOut`. 🤔

If that seems like a better name to you I'm certainly not opposed to switching to that.

Reply:
Just wanted to note explicitly that with this change none of [this code](https://github.com/Automattic/pocket-casts-android/blob/7a9fbad50b762563536f5b0f659117087d232832/modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/user/UserManager.kt#L108-L129) is getting run when the user clears data on sign-in:

``` kotlin
if (wasInitiatedByUser || !settings.getFullySignedOut()) {
    LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "Signing out")
    subscriptionManager.clearCachedStatus()
    syncManager.signOut {
        settings.clearPlusPreferences()
        GlobalScope.launch {
            userEpisodeManager.removeCloudStatusFromFiles(playbackManager)
        }

        settings.setMarketingOptIn(false)
        settings.setMarketingOptInNeedsSync(false)
        settings.setEndOfYearModalHasBeenShown(false)

        analyticsTracker.track(
            AnalyticsEvent.USER_SIGNED_OUT,
            mapOf(KEY_USER_INITIATED to wasInitiatedByUser)
        )
        analyticsTracker.flush()
        analyticsTracker.clearAllData()
        analyticsTracker.refreshMetadata()
    }
}

```

I think that's probably fine because all those things seem to relate to accounts and obviously the user isn't signed-in in this scenario, but it's worth keeping in mind if we see any issues.